lib/reek/smell_detectors/repeated_conditional.rb

```diff
@@ -74,13 +74,18 @@ def max_identical_ifs
       # @quality :reek:TooManyStatements { max_statements: 9 }
       def conditional_counts
         result = Hash.new { |hash, key| hash[key] = [] }
-        collector = proc do |node|
-          next unless (condition = node.condition)
-          next if condition == BLOCK_GIVEN_CONDITION
+        collector = proc do |method|
+          [:if, :case].map do |stmt|
+            method.each_node(stmt) do |node|
+              next unless (condition = node.condition)
+              next if condition == BLOCK_GIVEN_CONDITION
 
-          result[condition].push(condition.line)
+              result[method.name].push(condition.line)
```

Comment:
This is kind of incomplete. I was thinking of doing something like

```ruby
matches = {}
matches[method.name] = result[condition].push(condition.line)

{
  {:bravo => [4]},
  {:delta => [9]},
  {:echo => [12]},
}
```

Then count on each method if there are more that 2 or the allowed default